proto/expr.proto

```diff
@@ -141,7 +141,7 @@ message ExprNode {
   Type expr_type = 1;
   data.DataType return_type = 3;
   oneof rex_node {
-    InputRefExpr input_ref = 4;
+    uint32 input_ref = 4;
```

Comment:
Not sure whether this is a good change. 🤔 With a new type of `InputRefExpr`, the meaning of this unsigned integer seems more explicit.

Reply:
The rationale here is to uniformly use `uint32` as column index in proto. Previously we have `InputRefExpr`, but it's different from `InputRef` and `InputRefExpression` in Rust code. And we have another `Arg` type containing what `InputRef` has, which is confusing.

I tried to add a `ColumnIndex` type in proto, but found it hard to choose between `ColumnIndex` and `uint32` in many other proto message defs. E.g. we have `uint32 row_count_index` field referencing one column, we also have `repeated uint32 output_indices` referencing a vector of columns, if we use `ColumnIndex` for the former, should we also use it for the latter? If we do use it for all column index fields, then too many code need to be modified (and they will be `None`-able); if we don't, then we will face some inconsistency and `ColumnIndex` won't be able to hide the unsigned integer type.